error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/MockitoMockClassReferenceTest.java

```diff
@@ -0,0 +1,139 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode.TEXT_MATCH;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class MockitoMockClassReferenceTest {
+  private final CompilationTestHelper compilationTestHelper =
+      CompilationTestHelper.newInstance(MockitoMockClassReference.class, getClass());
+  private final BugCheckerRefactoringTestHelper refactoringTestHelper =
+      BugCheckerRefactoringTestHelper.newInstance(MockitoMockClassReference.class, getClass());
+
+  @Test
+  void identification() {
+    compilationTestHelper
+        .addSourceLines(
+            "A.java",
+            "import static org.mockito.Mockito.mock;",
+            "import static org.mockito.Mockito.spy;",
+            "",
+            "import java.util.List;",
+            "",
+            "class A {",
+            "  // BUG: Diagnostic contains:",
+            "  String memberMock = mock(String.class);",
+            "  // BUG: Diagnostic contains:",
+            "  String memberSpy = mock(String.class);",
+            "",
+            "  void m() {",
+            "    Integer variableMock;",
+            "    // BUG: Diagnostic contains:",
+            "    variableMock = mock(Integer.class);",
+            "    // BUG: Diagnostic contains:",
+            "    List nonGenericallyTypedMock = mock(List.class);",
+            "    var dynamicallyTypedMock = mock(Integer.class);",
+            "    Integer namedMock = mock(Integer.class, \"name\");",
+            "    Object subtypeMock = mock(Integer.class);",
+            "",
+            "    // BUG: Diagnostic contains:",
+            "    String equalTypedSpy = spy(String.class);",
+            "    // BUG: Diagnostic contains:",
+            "    List nonGenericallyTypedSpy = spy(List.class);",
+            "    var dynamicallyTypedSpy = spy(Integer.class);",
+            "    Object objectSpy = spy(new Object());",
+            "  }",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  void unimplementedCases() {
```

Comment:
I'd like some input as to how we wish to behave in these edge-cases and what would be the ideal way to detect those cases.